fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/ExpressionStatisticCalculator.java

```diff
@@ -58,11 +59,21 @@ public ColumnStatistic visitConstant(ConstantOperator operator, Void context) {
             if (value.isPresent()) {
                 return new ColumnStatistic(value.getAsDouble(), value.getAsDouble(), 0,
                         operator.getType().getSlotSize(), 1);
+            } else if (operator.getType().isStringType()) {
+                return new ColumnStatistic(Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY, 0, 1, 1);
             } else {
                 return ColumnStatistic.unknown();
             }
         }
 
+        @Override
+        public ColumnStatistic visitCaseWhenOperator(CaseWhenOperator caseWhenOperator, Void context) {
+            int whenClauseSize = caseWhenOperator.getWhenClauseSize();
+            int distinctValues = whenClauseSize + (caseWhenOperator.hasElse() ? 1 : 0);
+            return new ColumnStatistic(Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY, 0,
+                    caseWhenOperator.getType().getSlotSize(), distinctValues);
```

Comment:
The code looks will choose small distinct if then clause is a variables expression(column/expression), the modify can be only used on const expression.